### PR TITLE
Create extract-extensions-document.js

### PR DESCRIPTION
**Pull request by rs4231199**
Created a script that extracts extensions' names, and descriptions from [https://github.com/4ian/GDevelop-extensions/blob/master/extensions-registry.json](https://github.com/4ian/GDevelop-extensions/blob/master/extensions-registry.json) file and write that into a markdown file.

### CODE REVIEW

**Comment by arthuro555** [comment]
This script should probably be on the GDevelop-extensions repo instead of this one.

**Comment by rs4231199** [comment]
I already have discussed this with @4ian . He said to put it here.

File: newIDE/app/scripts/extract-extensions-document.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+const writeFile = content => {
+  return new Promise((resolve, reject) => {
+    fs.writeFile(outputFile, content, err => {
+      if (err) return reject(err);
+      resolve();
+    });
+  });
+};
+
+https.get(extensionsRegistoryURL, function(res) {
```
Comment (4ian) [inline newIDE/app/scripts/extract-extensions-document.js:20]: While it's true that your solution works, I now recommend using `axios` (as done in extract-changelog.js): 

```
const axios = require('axios');

axios.get(extensionsRegistryUrl).then(response => ...)
```

Or even better using async/await :)

File: newIDE/app/scripts/extract-extensions-document.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  let data = '',
+    json_data;
```
Comment (4ian) [inline newIDE/app/scripts/extract-extensions-document.js:22]: nitpicking: don't declare early variables.

File: newIDE/app/scripts/extract-extensions-document.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+  res.on('data', function(stream) {
+    data += stream;
+  });
+  
+  res.on('end', function() {
+    json_data = JSON.parse(data);
```
Comment (4ian) [inline newIDE/app/scripts/extract-extensions-document.js:29]: nitpicking: camelCase: `jsonData`

Also try/catch around the JSON.parse

**Comment by 4ian** [comment]
Looks mostly good! Added a few comments. Thanks for working on this 👍

**Comment by 4ian** [comment]
Great! Merging now :) I've run Prettier and fixed the headings. 
The resulting page is here: http://wiki.compilgames.net/doku.php/gdevelop5/extensions/list

Ideally we could maybe run this like the documentation, as a regular task run on a CI like Circle CI that then upload the resulting files to the wiki hosting (cc @arthuro555), and we could run also extract-reference-document.js for the expression reference (http://wiki.compilgames.net/doku.php/gdevelop5/all-features/expressions-reference). I could give access to CircleCI/the Github repo of GDevelop-extensions to do that, like done here: https://github.com/4ian/GDevelop-documentation/blob/master/.circleci/config.yml

**Comment by rs4231199** [comment]
@4ian. Hey Florian. You CCed arthuro555 in your message so I assumed my work here was done. Just to confirm the same and if there is any work for me I would be delighted to do that.

**Comment by 4ian** [comment]
Nothing more currently, I'm thinking of how to automate the calls to extract-reference-document.js and extract-extensions-document.js and upload the result to the wiki :) 
I think a regular CircleCI workflow like done for the documentation, with a `scp` to the server hosting the wiki would work.

**Pull request merged**